integration-cli/docker_api_netmode_test.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"os/exec"
+	"strings"
+
+	"github.com/docker/docker/runconfig"
+	"github.com/go-check/check"
+)
+
+// GH14530. Validates combinations of --net= with other options
+func checkPS(out string) bool {
+	return strings.Contains(out, "PID   USER")
+}
+
+func (s *DockerSuite) TestNetHostname(c *check.C) {
+
+	var (
+		out    string
+		err    error
+		runCmd *exec.Cmd
+	)
+
+	runCmd = exec.Command(dockerBinary, "run", "-h=name", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err != nil {
+		c.Fatalf(out, err)
+	}
+	if !checkPS(out) {
+		c.Fatalf("Expected PS output, got %s", out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "--net=host", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err != nil {
+		c.Fatalf(out, err)
+	}
+	if !checkPS(out) {
+		c.Fatalf("Expected PS output, got %s", out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "-h=name", "--net=bridge", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err != nil {
+		c.Fatalf(out, err)
+	}
+	if !checkPS(out) {
+		c.Fatalf("Expected PS output, got %s", out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "-h=name", "--net=none", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err != nil {
+		c.Fatalf(out, err)
+	}
+	if !checkPS(out) {
+		c.Fatalf("Expected PS output, got %s", out)
+	}
+
+	runCmd = exec.Command(dockerBinary, "run", "-h=name", "--net=host", "busybox", "ps")
+	if out, _, err = runCommandWithOutput(runCmd); err == nil {
+		c.Fatalf(out, err)
```

Comment:
this should be `err == nil || !strings.Contains(...) {` so that the Fatal message has information on what's expected. Same for all the other `err == nil` checks below